src/mono/wasm/debugger/DebuggerTestSuite/AssignmentTests.cs

```diff
@@ -0,0 +1,66 @@
+            { "MONO_TYPE_STRING",      TString(default),                                        TString("hello") },
+            { "MONO_TYPE_ENUM",        TEnum("DebuggerTests.RGB", "Red"),                       TEnum("DebuggerTests.RGB", "Blue") },
+            { "MONO_TYPE_ARRAY",       TObject("byte[]", is_null: true),                        TArray("byte[]", 2) },
+            { "MONO_TYPE_VALUETYPE",   TValueType("DebuggerTests.Point"),                       TValueType("DebuggerTests.Point") },
+            { "MONO_TYPE_VALUETYPE2",  TValueType("System.Decimal","0"),                        TValueType("System.Decimal", "1.1") },
+            { "MONO_TYPE_GENERICINST", TObject("System.Func<int>", is_null: true),              TDelegate("System.Func<int>", "int Prepare ()") },
+            { "MONO_TYPE_FNPTR",       TPointer("*()",  is_null: true),                         TPointer("*()") },
+            { "MONO_TYPE_PTR",         TPointer("int*", is_null: true),                         TPointer("int*") },
+            { "MONO_TYPE_I1",          TNumber(0),                                              TNumber(-1) },
+            { "MONO_TYPE_I2",          TNumber(0),                                              TNumber(-1) },
+            { "MONO_TYPE_I4",          TNumber(0),                                              TNumber(-1) },
+            { "MONO_TYPE_I8",          TNumber(0),                                              TNumber(-1) },
+            { "MONO_TYPE_U1",          TNumber(0),                                              TNumber(1) },
+            { "MONO_TYPE_U2",          TNumber(0),                                              TNumber(1) },
+            { "MONO_TYPE_U4",          TNumber(0),                                              TNumber(1) },
+            { "MONO_TYPE_U8",          TNumber(0),                                              TNumber(1) },
+            { "MONO_TYPE_R4",          TNumber(0),                                              TNumber("3.1414999961853027") },
+            { "MONO_TYPE_R8",          TNumber(0),                                              TNumber("3.1415") },
+        };
+
+        [Theory]
+        [MemberData("GetTestData")]
+        async Task InspectVariableBeforeAndAfterAssignment(string clazz, JObject checkDefault, JObject checkValue)
+        {
+            await SetBreakpointInMethod("debugger-test", "DebuggerTests." + clazz, "Prepare", 2);
+            await EvaluateAndCheck("window.setTimeout(function() { invoke_static_method('[debugger-test] DebuggerTests." + clazz + ":Prepare'); })", null, -1, -1, "Prepare");
+
+            // 1) check un-assigned variables
+            await StepAndCheck(StepKind.Into, "dotnet://debugger-test.dll/debugger-assignment-test.cs", -1, -1, "TestedMethod",
+                locals_fn: (locals) =>
+                {
+                    Assert.Equal(2, locals.Count());
+                    Check(locals, "r", checkDefault);
```

Comment:
Instead of the new overloads that use `CheckValue`, you can use:
```csharp
await CheckProps(locals, new
{
  r = checkDefault
}, "default", num_fields: 2);
```

But since this needs to be async, you will need to do:
```csharp
var pause_location = await StepAndCheck(StepKind.Into, "dotnet://debugger-test.dll/debugger-assignment-test.cs", -1, -1, "TestedMethod");
var locals = await GetProperties(pause_location["callFrames"][0]["callFrameId"].Value<string>());

await CheckProps(locals, new
{
  r = checkDefault
}, "default", num_fields: 2);
```